src/Api/FingerprintApi.php

```diff
@@ -1,739 +1,142 @@
-
-        try {
-            $options = $this->createHttpClientOption();
-            try {
-                $response = $this->client->send($request, $options);
-            } catch (RequestException $e) {
-                throw new ApiException(
-                    "[{$e->getCode()}] {$e->getMessage()}",
-                    $e->getCode(),
-                    $e->getResponse() ? $e->getResponse()->getHeaders() : null,
-                    $e->getResponse() ? $e->getResponse()->getBody()->getContents() : null
-                );
-            }
-
-            $statusCode = $response->getStatusCode();
-
-            if ($statusCode < 200 || $statusCode > 299) {
-                throw new ApiException(
-                    sprintf(
-                        '[%d] Error connecting to the API (%s)',
-                        $statusCode,
-                        $request->getUri()
-                    ),
-                    $statusCode,
-                    $response->getHeaders(),
-                    $response->getBody()
-                );
-            }
-
-            $responseBody = $response->getBody();
-            if ($returnType === '\SplFileObject') {
-                $content = $responseBody; //stream goes to serializer
-            } else {
-                $content = $responseBody->getContents();
-                if (!in_array($returnType, ['string','integer','bool'])) {
-                    $content = json_decode($content);
-                }
-            }
-
-            return [
-                ObjectSerializer::deserialize($content, $returnType, []),
-                $response->getStatusCode(),
-                $response->getHeaders()
-            ];
-
-        } catch (ApiException $e) {
-            switch ($e->getCode()) {
-                case 200:
-                    $data = ObjectSerializer::deserialize(
-                        $e->getResponseBody(),
-                        '\Fingerprint\ServerAPI\Model\EventResponse',
-                        $e->getResponseHeaders()
-                    );
-                    $e->setResponseObject($data);
-                    break;
-                case 403:
-                    $data = ObjectSerializer::deserialize(
-                        $e->getResponseBody(),
-                        '\Fingerprint\ServerAPI\Model\ErrorEvent403Response',
-                        $e->getResponseHeaders()
-                    );
-                    $e->setResponseObject($data);
-                    break;
-                case 404:
-                    $data = ObjectSerializer::deserialize(
-                        $e->getResponseBody(),
-                        '\Fingerprint\ServerAPI\Model\ErrorEvent404Response',
-                        $e->getResponseHeaders()
-                    );
-                    $e->setResponseObject($data);
-                    break;
-            }
-            throw $e;
-        }
+        return $this->config->getHost() . "/events/{request_id}";
```

Comment:
Suggestion: consider looking at how the re-write was implemented in Go or Dotnet SDKs.
The main idea is to get as much data as possible from the schema, rather than hard-coding it.  We can easily extract the endpoints from schema instead.
Right now in case of new endpoint, or an change in existing one we will have to remember to update it manually here.

For reference, take a look at these files from GO SDK:
- template file for API that includes simple struct for every API operation (which includes the endpoint): https://github.com/fingerprintjs/fingerprint-pro-server-api-go-sdk/blob/main/template/api.mustache
- generated file with these structs: https://github.com/fingerprintjs/fingerprint-pro-server-api-go-sdk/blob/main/sdk/api_fingerprint.go
- how requests are being made: https://github.com/fingerprintjs/fingerprint-pro-server-api-go-sdk/blob/main/sdk/api_fingerprint_impl.go (this part is written manually)

Reply:
Can you recheck this part to see if it's consistent with other SDKs? If not, what is the clear expectation to change in this repository?